dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -74,11 +88,61 @@ static std::wstring g_test_ddlmPackagePublisherId;
 STDAPI MddBootstrapInitialize(
     UINT32 majorMinorVersion,
     PCWSTR versionTag,
-    PACKAGE_VERSION minVersion) noexcept try
+    PACKAGE_VERSION minVersion) noexcept
+{
+    return MddBootstrapInitialize2(majorMinorVersion, versionTag, minVersion, MddBootstrapInitializeOptions_None);
+}
+
+STDAPI MddBootstrapInitialize2(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion,
+    MddBootstrapInitializeOptions options) noexcept try
 {
     // Dynamic Dependencies Bootstrap API requires a non-packaged process
     LOG_HR_IF(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED), AppModel::Identity::IsPackagedProcess());
 
+    // Are we already initialized?
+    const auto hr{ _MddBootstrapInitialize(majorMinorVersion, versionTag, minVersion) };
+    if (FAILED(hr))
+    {
+        LOG_IF_FAILED(MddBootstrapInitialize_Log(hr, majorMinorVersion, versionTag, minVersion));
+
+        if (WI_IsFlagSet(options, MddBootstrapInitializeOptions_OnError_DebugBreak) ||
+            (WI_IsFlagSet(options, MddBootstrapInitializeOptions_OnError_DebugBreak_IfDebuggerAttached) && IsDebuggerPresent()) ||
+            IsOptionEnabled(L"MICROSOFT_WINDOWSAPPRUNTIME_BOOTSTRAP_INITIALIZE_DEBUGBREAK"))
+        {
+            DebugBreak();
+        }
+
+        if (WI_IsFlagSet(options, MddBootstrapInitializeOptions_OnNoMatch_ShowUI) ||
```

Comment:
are there other HRESULTs that we can expect, besides ERROR_NO_MATCH?  if so, rename enum to MddBootstrapInitializeOptions_OnError_ShowUI

Reply:
It feels brittle to tie the behavior to specific HRESULTs, which would be all-too-easy to change in future.  So I'd opt for 2, with the somewhat weasily language.

Reply:
enum name still doesn't match behavior, which is to pop the UI for all HRESULTs

Reply:
Good point. Mulled it over for a bit and I think the enum is right / fix the code, for 1.1, and consider the other-errors case for 1.2 different message, different URL (https://aka.ms/windowsappsdk/1.1/troubleshooting ?)

Code updated. Thanks!